src/explorers/ethereum/etherscan.js

```diff
@@ -1,44 +1,26 @@
-import { request } from '../../services';
-import { BLOCKCHAINS, CONFIG, SUB_STEPS, TRANSACTION_APIS } from '../../constants';
+import { request } from '../../services/request';
+import { BLOCKCHAINS, CONFIG, SUB_STEPS, TRANSACTION_ID_PLACEHOLDER } from '../../constants';
 import { generateTransactionData, VerifierError } from '../../models';
 import { stripHashPrefix } from '../utils/stripHashPrefix';
 import { getText } from '../../domain/i18n/useCases';
-import { TRANSACTIONS_APIS_URLS } from '../../constants/api';
+import { buildTransactionServiceUrl } from '../../services/transaction-apis';
+import { isTestChain } from '../../constants/blockchains';
 
-export function getEtherScanFetcher (transactionId, chain) {
-  const action = '&action=eth_getTransactionByHash&txhash=';
-  let etherScanUrl;
-  if (chain === BLOCKCHAINS.ethmain.code) {
-    etherScanUrl = TRANSACTIONS_APIS_URLS[TRANSACTION_APIS.Etherscan].main + action + transactionId;
-  } else {
-    etherScanUrl = TRANSACTIONS_APIS_URLS[TRANSACTION_APIS.Etherscan].test + action + transactionId;
-  }
-
-  const etherScanFetcher = new Promise((resolve, reject) => {
-    return request({ url: etherScanUrl })
-      .then(function (response) {
-        const responseTxData = JSON.parse(response);
-        try {
-          // Parse block to get timestamp first, then create generateTransactionData
-          const blockFetcher = getEtherScanBlock(responseTxData, chain);
-          blockFetcher
-            .then(function (blockResponse) {
-              const txData = parseEtherScanResponse(responseTxData, blockResponse);
-              resolve(txData);
-            })
-            .catch(function () {
-              reject(new VerifierError(SUB_STEPS.fetchRemoteHash, getText('errors', 'unableToGetRemoteHash')));
-            });
-        } catch (err) {
-          // don't need to wrap this exception
-          reject(new VerifierError(SUB_STEPS.fetchRemoteHash, getText('errors', 'unableToGetRemoteHash')));
-        }
-      }).catch(function () {
-        reject(new VerifierError(SUB_STEPS.fetchRemoteHash, getText('errors', 'unableToGetRemoteHash')));
-      });
-  });
-  return etherScanFetcher;
-}
+const ETHERSCAN_API_KEY = 'FJ3CZWH8PQBV8W5U6JR8TMKAYDHBKQ3B1D';
```

Comment:
we need to account somewhere, and I think that's what we forgot in #471, that you could pass your own keys for the explorers that are defined by default. It's not related to this PR, but we should not forget

Reply:
I first thought we would leave Etherscan parsing function in the explorers folder, so users could copy / paste it in their implementation. But we can open the API to send keys for the public explorer APIs we support. I'll come up with a model.

Reply:
I think let's wait until we have the injection working. Things might be clearer then. But I agree, the explorer function should remain in the repo.

Reply:
I agree.